controller/src/controller.rs

```diff
@@ -379,6 +413,66 @@ fn node_has_label(node: &Node) -> bool {
         ));
 }
 
+#[instrument(skip())]
+fn within_time_window() -> Result<bool> {
+    let prevent_update_since_env = env::var(PREVENT_UPDATE_SINCE_ENV_VAR).context(
+        controllerclient_error::MissingEnvVariableSnafu {
+            variable: PREVENT_UPDATE_SINCE_ENV_VAR.to_string(),
+        },
+    )?;
+
+    let prevent_update_until_env = env::var(PREVENT_UPDATE_UNTIL_ENV_VAR).context(
+        controllerclient_error::MissingEnvVariableSnafu {
+            variable: PREVENT_UPDATE_UNTIL_ENV_VAR.to_string(),
+        },
+    )?;
+
+    let current_time = Utc::now().time();
+    let prevent_update_since = NaiveTime::parse_from_str(&prevent_update_since_env, "%H:%M:%S")
+        .context(controllerclient_error::ConvertToNativeTimeSnafu {
+            date: prevent_update_since_env,
+        })?;
+    let prevent_update_until = NaiveTime::parse_from_str(&prevent_update_until_env, "%H:%M:%S")
+        .context(controllerclient_error::ConvertToNativeTimeSnafu {
+            date: prevent_update_until_env,
+        })?;
+
+    // Due to the situation that controller has already started a node update but the time window is coming,
+    // we design controller to stop updating any new nodes 6 mins (the time brupop spends to update a node)
```

Comment:
Should add this behavior to the docs so users aren't surprised when there's a potential 6 minute buffer between their window and an occurring update.

Reply:
Sure! I'll add it.